tests/integration/bond_test.py

```diff
@@ -553,3 +555,42 @@ def test_set_miimon_100_on_existing_bond(bond99_with_2_slaves):
     bond_config[Bond.OPTIONS_SUBTREE] = {"miimon": 100}
     libnmstate.apply(state)
     assertlib.assert_state_match(state)
+
+
+@pytest.fixture
+def eth1_eth2_with_no_profile():
+    iface_down(ETH1)
+    iface_down(ETH2)
+    yield
+    iface_down(ETH1)
+    iface_down(ETH2)
+
+
+def test_bond_mac_be_consistant_with_diff_slave_order(
+    eth1_eth2_with_no_profile,
+):
+    """
+    When sever reboot, network manager will by default activate slaves in the
```

Comment:
How can we test the behavior after reboot?

Reply:
Maybe just deactivate all the slaves and bond and activate the bond via `nmcli`.
That could simulate the reboot process.

Reply:
Let me see how can I do that in test case.

Reply:
@thom311 is it possible to deactivate all profiles and then to restart NM to simulate a system boot? If not, how could it be simulated in a container?

Reply:
In VDSM we use this NM config: https://github.com/oVirt/vdsm/blob/master/static/etc/NetworkManager/conf.d/vdsm.conf#L5

Reply:
Test case updated using `nmcli` to simulator the server reboot.

Reply:
Please note my comments regarding the excpected scenario as defined in https://bugzilla.redhat.com/show_bug.cgi?id=1812386#c3

Reply:
> In VDSM we use this NM config: https://github.com/oVirt/vdsm/blob/master/static/etc/NetworkManager/conf.d/vdsm.conf#L5

What is your suggestion for Nmstate to do here?

Reply:
> In VDSM we use this NM config: https://github.com/oVirt/vdsm/blob/master/static/etc/NetworkManager/conf.d/vdsm.conf#L5

That's the default action of NetworkManager and the other option `order by kernel index` make no sense that any one would use it.

Reply:
I have left a comment in bug 1812386 with summery of my finding on this.

Reply:
Actually it is not the default, at least it was not the default when we added it.
We added it to VDSM when we saw the problem with the NM default as I recall.
(it broke consistency with initscripts)

Can you re-confirm this with the NM team?

Reply:
> is it possible to deactivate all profiles and then to restart NM to simulate a system boot? If not, how could it be simulated in a container?

If you stop NM, delete all files in /run/NetworkManager, flush kernel configuration from devices (addresses, routes, master-slave relationships) and start NM again, that should be equivalent to a fresh start after a reboot.

Reply:
The order in which NM activates slaves (during boot or when autoconnecting slaves) is 'name' by default. But on RHEL7 only, there is a configuration snippet in /usr/lib/ that changes the default to 'ifindex' for backwards compatibility.

Reply:
the build in default is `name`.

But on RHEL-7 (only), NM packages a file `/usr/lib/NetworkManager/conf.d/10-slaves-order.conf`, which sets the value to `slave-order=ifindex`. This is to preserve previous behavior on rhel-7.

Reply:
@bengal Is that sufficient to just `nmcli c down bond99` and up again?

I think the kernel configure will be flush for bond99 and its slaves.
For `/run/NetworkManager`, don't know that impact on this MAC address choosing.

Reply:
If you want to test the order of autoconnect-slaves during an explicit activation, yes, a `nmcli connection down` followed by up is sufficient.

Reply:
The procedure I described above to delete /run/NM clears all the previous state and so it is useful to simulate a fresh start after a machine boot.

Reply:
Thanks.